### ISSUE TITLE
Invalid HTML rendered in TOC, when URLs appear in titles

### ISSUE THREAD

**Issue by davidfavor** [asciidoctor/asciidoctor#1282]
If a title appears in a document of the form...

```
2001: http://LiveFeast.com was born
```

The TOC generation produces invalid HTML of the form...

```
<li><a href="#_2001_a_href_http_livefeast_com_class_bare_http_livefeast_com_a_was_born">2001: <a href="http://LiveFeast.com     " class="bare">http://LiveFeast.com</a> was born</a></li>
```

Notice there at nested `<a>` + `</a>` tags in the TOC.

Likely there's some general approach to getting around this.

So this is not really a bug report, more a question about the correct way to generate TOCs with clickable links + valid HTML.

Someone, let me know how this is best handled.

Thanks.

**Comment by davidfavor** [comment]
Oh... My toolchain is...

asciidoctor -a year=2015 -d book --backend html5 -a encoding=UTF-8 -a toc -a toclevels=6 -e erubis -o out in

**Comment by mojavelinux** [comment]
I actually consider this to be a bug (or at least a deficiency in Asciidoctor that we should fix). I've seen this before and it isn't the behavior I expect. I think that in the TOC, the title should be displayed without links because the TOC itself is a link. We should perhaps strip them out when the TOC is displayed.

I'll schedule this for the 1.5.4 release, but it might make a good task for a conference hack session.

**Comment by atbest** [comment]
This bug is still present in current version 1.5.5. In addition to the TOC text, I think the url information show be stripped in the auto-generated href attributes of headers.

**Comment by atbest** [comment]
I found a simple method to strip the html tags by modifying lines 306-310 in html5.rb:
From
```ruby
          result << %(<li><a href="##{section.id}">#{section_num}#{section.captioned_title}</a>)
          result << child_toc_level
          result << '</li>'
        else
          result << %(<li><a href="##{section.id}">#{section_num}#{section.captioned_title}</a></li>)
```
To
```ruby
          result << %(<li><a href="##{section.id}">#{section_num}#{section.captioned_title.gsub(/<.*?>/, '')}</a>)
          result << child_toc_level
          result << '</li>'
        else
          result << %(<li><a href="##{section.id}">#{section_num}#{section.captioned_title.gsub(/<.*?>/, '')}</a></li>)
```
The gsub will remove all html tags from `section.captioned_title` and only plain text is exported, which should be the correct behavior for toc.

**Comment by mojavelinux** [comment]
There's actually a method in the Asciidoctor API that does something similar. See https://github.com/asciidoctor/asciidoctor/blob/2796abffbf7fbd32d46cd9df0bebff6ea0da7841/lib/asciidoctor/document.rb#L47-L49 We just need to extract it into a general purpose utility so we can use it here too.

**Comment by atbest** [comment]
Yes, you can replace `.gsub(/<.*?>/, '')` with `.gsub(XmlSanitizeRx, '')`.

**Comment by mojavelinux** [comment]
This will be resolved along with #840.

**Comment by mojavelinux** [comment]
This was resolved by ce927bdf045f6d176e0c173e67654786e7abbce9.
